src/sequence.rs

```diff
@@ -0,0 +1,155 @@
+        Err(e) => {
+            // Remove copied music file (clean up)
+            try!(fs::remove_file(&dest_path).map_err(Error::Io));
+            panic!(e.to_string())
+        },
+    };
+
+    // Save project
+    try!(utils::write_protonfile(&new_project, None::<P>));
+
+    // Commit changes
+    let signature = Signature::now("Proton Lights", "proton@teslaworks.net").unwrap();
+    let msg = format!("Adding new sequence '{}'", name);
+    let repo_path: Option<P> = None;
+
+    utils::commit_all(repo_path, &signature, &msg)
+        .map(|_| ())
+}
+
+/// Check that the music file is a valid format
+/// Full list of supported formats can be found at
+/// http://www.rust-sfml.org/doc/rsfml/audio/struct.Music.html
+fn verify_file_type<P: AsRef<Path>>(music_file_path: P) -> Result<(), Error> {
+    match music_file_path.as_ref().extension() {
+        Some(extension) => {
+            match extension.to_str() {
+                Some("ogg")  |
+                Some("wav")  |
+                Some("flac") |
+                Some("aiff") |
+                Some("raw") => Ok(()),
+                None => Err(unsupported_file_type_error("invalid extension")),
+                _ => Err(unsupported_file_type_error(extension.to_str().unwrap())),
+            }
+        },
+        None => Err(unsupported_file_type_error("unknown")),
+    }
+}
+
+/// Makes sure the name has only valid characters in it
+/// A valid character is upper and lower alpha, numbers, and underscores
+fn verify_name(name: &str) -> Result<(), Error> {
+    for c in name.chars() {
+        let alpha_lower = 'a' <= c && c <= 'z';
+        let alpha_upper = 'A' <= c && c <= 'Z';
+        let number = '0' <= c && c <= '9';
+        if !(alpha_lower || alpha_upper || number || c == '_') {
+            return Err(invalid_sequence_name_error(name));
+        }
+    }
+
+    Ok(())
+}
+
+/// Copies the file at music_file_path to the current directory
+/// Throw error if file does not exist
+///
+/// Impure.
+fn copy_music_file<P: AsRef<Path>>(music_file_path: P, dest_folder: &str) -> Result<PathBuf, Error> {
+    // Make sure source file exists
+    if !music_file_path.as_ref().exists() {
+        Err(music_file_not_found_error(music_file_path))
+    } else {
+        let file_name = try!(utils::file_name_from_path(&music_file_path));
+        let dest_path = Path::new(&dest_folder).join(&file_name);
+        try!(fs::copy(&music_file_path, &dest_path)
+            .map_err(Error::Io)
+            .map(|_| Ok(PathBuf::from(dest_path))))
+        
+
+    }
+
+}
+
+/// Extracts the duration of a music file
+/// Wraps errors in Error::Rsfml
+fn get_music_duration_sec(path: &str) -> Result<u32, Error> {
+    let music = match Music::new_from_file(&path) {
+        Some(m) => m,
+        None => return Err(rsfml_error("Error reading file.")),
+    };
+    let duration_time = music.get_duration();
+    let duration = duration_time.as_seconds() as u32;
+    Ok(duration)
```

Comment:
if you mean `Ok(duration_time.as_seconds() as u32);` I tried it now and Rust wasn't able to infer type info. That `let` might be necessary.